### PR TITLE
Fixes #6032 and #5942

### LINKED ISSUE #6032: [TEST-FAILURE] lock.ConditionBasicDistributedTest.testSignalWithSingleWaiter
```
java.lang.AssertionError: expected:<false> but was:<true>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at com.hazelcast.concurrent.lock.ConditionBasicTest.testSignalWithSingleWaiter(ConditionBasicTest.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
```

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-3.x/com.hazelcast$hazelcast/3637/testReport/junit/com.hazelcast.concurrent.lock/ConditionBasicDistributedTest/testSignalWithSingleWaiter/

### PR DESCRIPTION
**Pull request by tombujok**
Fixes #6032 and #5942
* All risky sleeps removed from the tests
* Execution flow orchestrated by latches and locks

### CODE REVIEW

**Comment by gurbuzali** [comment]
looks good

File: hazelcast/src/test/java/com/hazelcast/test/HazelcastTestSupport.java [hunk 1]
```diff
@@ -735,6 +736,48 @@ public static void assertOpenEventually(String message, CountDownLatch latch, lo
...
+                    if (atomic.get() != expectedValue) {
+                        throw new AssertionError("Atomic value has not been met. " + message);
+                    }
+                    sleepMillis(50);
+                }
+            }
+        }, timeoutInSeconds);
+    }
+
+    public static void assertAtomicEventually(final String message, final AtomicBoolean atomic, final boolean expectedValue, int timeoutInSeconds) {
```
Comment (jerrinot) [inline hazelcast/src/test/java/com/hazelcast/test/HazelcastTestSupport.java]: just a minor one - the expected value is traditionally before the actual value in an argument list.

Comment (tombujok) [inline hazelcast/src/test/java/com/hazelcast/test/HazelcastTestSupport.java]: Thx. Fixed. 

P.S. Oh, I hate traditions like this - that's why I love assertj ;)

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**